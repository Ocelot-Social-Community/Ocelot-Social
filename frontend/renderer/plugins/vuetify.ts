// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
  })
