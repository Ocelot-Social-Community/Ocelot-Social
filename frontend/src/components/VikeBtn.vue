<template>
  <v-btn
    :variant="isRouteSelected($attrs.href as string) ? 'tonal' : 'flat'"
    @click.prevent="onClick($attrs.href as string)"
  >
    <slot />
  </v-btn>
</template>
<script lang="ts" setup>
import { navigate } from 'vike/client/router'

import { usePageContext } from '#context/usePageContext'

const pageContext = usePageContext()

function onClick(href: string) {
  return navigate(href)
}

const isRouteSelected = (href: string) => {
  if (href === '/app') {
    return pageContext.urlPathname.indexOf(href) === 0
  }
  return pageContext.urlPathname === href
}
</script>
