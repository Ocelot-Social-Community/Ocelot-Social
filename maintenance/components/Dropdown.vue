<template>
  <v-popover
    v-model:open="isPopoverOpen"
    :open-group="Math.random().toString()"
    :placement="placement"
    :disabled="disabled"
    trigger="manual"
    :offset="offset"
  >
    <slot :toggle-menu="toggleMenu" :open-menu="openMenu" :close-menu="closeMenu" :is-open="isOpen" />
    <template #popover>
<div  @mouseover="popoverMouseEnter" @mouseleave="popoverMouseLeave">
      <slot
        name="popover"
        :toggle-menu="toggleMenu"
        :open-menu="openMenu"
        :close-menu="closeMenu"
        :is-open="isOpen"
      />
    </div>
</template>
  </v-popover>
</template>

<script>
let mouseEnterTimer = null
let mouseLeaveTimer = null

export default {
  props: {
    placement: { type: String, default: 'bottom-end' },
    disabled: { type: Boolean, default: false },
    offset: { type: [String, Number], default: '16' },
  },
  data() {
    return {
      isPopoverOpen: false,
      developerNoAutoClosing: false, // stops automatic closing of menu for developer purposes: default is 'false'
    }
  },
  computed: {
    isOpen() {
      return this.isPopoverOpen
    },
  },
  watch: {
    isPopoverOpen: {
      immediate: true,
      handler(isOpen) {
        try {
          if (isOpen) {
            this.$nextTick(() => {
              setTimeout(() => {
                const paddingRightStyle = `${
                  window.innerWidth - document.documentElement.clientWidth
                }px`
                const navigationElement = document.querySelector('.main-navigation')
                document.body.style.paddingRight = paddingRightStyle
                document.body.classList.add('dropdown-open')
                if (navigationElement) {
                  navigationElement.style.paddingRight = paddingRightStyle
                }
              }, 20)
            })
          } else {
            const navigationElement = document.querySelector('.main-navigation')
            document.body.style.paddingRight = null
            document.body.classList.remove('dropdown-open')
            if (navigationElement) {
              navigationElement.style.paddingRight = null
            }
          }
        } catch (err) {}
      },
    },
  },
  beforeUnmount() {
    clearTimeout(mouseEnterTimer)
    clearTimeout(mouseLeaveTimer)
  },
  methods: {
    toggleMenu() {
      this.isPopoverOpen ? this.closeMenu(false) : this.openMenu(false)
    },
    openMenu(useTimeout) {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (useTimeout === true) {
        this.popoverMouseEnter()
      } else {
        this.isPopoverOpen = true
      }
    },
    closeMenu(useTimeout) {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (useTimeout === true) {
        this.popoverMouseLeave()
      } else {
        this.isPopoverOpen = false
      }
    },
    popoverMouseEnter() {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (!this.isPopoverOpen) {
        mouseEnterTimer = setTimeout(() => {
          this.isPopoverOpen = true
        }, 500)
      }
    },
    popoverMouseLeave() {
      if (this.developerNoAutoClosing) return
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (this.isPopoverOpen) {
        mouseLeaveTimer = setTimeout(() => {
          this.isPopoverOpen = false
        }, 300)
      }
    },
    clearTimeouts() {
      clearTimeout(mouseEnterTimer)
      clearTimeout(mouseLeaveTimer)
    },
  },
}
</script>
