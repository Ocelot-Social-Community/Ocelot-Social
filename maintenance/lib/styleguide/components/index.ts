import Avatar from './data-display/Avatar/Avatar.vue'
import CopyField from './data-display/CopyField/CopyField.vue'
import List from './data-display/List/List.vue'
import Number from './data-display/Number/Number.vue'
import Table from './data-display/Table/Table.vue'
import Form from './data-input/Form/Form.vue'
import FormItem from './data-input/FormItem/FormItem.vue'
import Input from './data-input/Input/Input.vue'
import Radio from './data-input/Radio/Radio.vue'
import Select from './data-input/Select/Select.vue'
import Container from './layout/Container/Container.vue'
import Card from './layout/Card/Card.vue'
import Grid from './layout/Grid/Grid.vue'
import Modal from './layout/Modal/Modal.vue'
import Page from './layout/Page/Page.vue'
import PageTitle from './layout/PageTitle/PageTitle.vue'
import Placeholder from './layout/Placeholder/Placeholder.vue'
import Section from './layout/Section/Section.vue'
import Spinner from './layout/Spinner/Spinner.vue'
import Button from './navigation/Button/Button.vue'
import Menu from './navigation/Menu/Menu.vue'
import Chip from './typography/Chip/Chip.vue'
import Code from './typography/Code/Code.vue'
import Heading from './typography/Heading/Heading.vue'
import Icon from './typography/Icon/Icon.vue'
import Logo from './typography/Logo/Logo.vue'
import Tag from './typography/Tag/Tag.vue'
import Text from './typography/Text/Text.vue'

export {
  Avatar,
  CopyField,
  List,
  Number,
  Table,
  Form,
  FormItem,
  Input,
  Radio,
  Select,
  Container,
  Card,
  Grid,
  Modal,
  Page,
  PageTitle,
  Placeholder,
  Section,
  Spinner,
  Button,
  Menu,
  Chip,
  Code,
  Heading,
  Icon,
  Logo,
  Tag,
  Text,
}
