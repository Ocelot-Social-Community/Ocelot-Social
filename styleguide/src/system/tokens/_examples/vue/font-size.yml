#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
