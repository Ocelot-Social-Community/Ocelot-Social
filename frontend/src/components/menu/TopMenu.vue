<template>
  <v-app-bar flat>
    <v-row>
      <v-col>
        <LogoAvatar />
      </v-col>
      <v-col class="d-flex align-center justify-center grow">
        <VikeBtn href="/">{{ $t('menu.home') }}</VikeBtn>
        <VikeBtn href="/app">{{ $t('menu.app') }}</VikeBtn>
        <VikeBtn href="/about">{{ $t('menu.about') }}</VikeBtn>
      </v-col>
      <v-col>
        <v-switch
          v-model="isEnabled"
          class="d-flex justify-end mr-5"
          :label="$t('language.german')"
          color="success"
          @update:model-value="onChange"
        ></v-switch>
      </v-col>
    </v-row>
  </v-app-bar>
</template>

<script lang="ts" setup>
import { ref } from 'vue'
import { useLocale } from 'vuetify'

import VikeBtn from '#components/VikeBtn.vue'

import LogoAvatar from './LogoAvatar.vue'

const { current: locale } = useLocale()
const isEnabled = ref(locale.value === 'de')
const onChange = () => {
  locale.value = isEnabled.value ? 'de' : 'en'
}
</script>
