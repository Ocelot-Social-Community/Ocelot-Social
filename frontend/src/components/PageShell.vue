<template>
  <v-app>
    <TopMenu />
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
