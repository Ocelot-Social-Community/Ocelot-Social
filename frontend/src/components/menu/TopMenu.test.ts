import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import { h } from 'vue'
import { VApp } from 'vuetify/components'

import TopMenu from './TopMenu.vue'

describe('TopMenu', () => {
  const wrapper = mount(VApp, {
    slots: {
      default: h(TopMenu),
    },
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
