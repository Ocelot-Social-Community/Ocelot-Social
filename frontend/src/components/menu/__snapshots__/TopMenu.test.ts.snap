// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopMenu > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    <header
      class="v-toolbar v-toolbar--flat v-toolbar--density-default v-theme--light v-locale--is-ltr v-app-bar"
      style="top: 0px; z-index: 1004; transform: translateY(0%); position: fixed; left: 0px; width: calc(100% - 0px - 0px);"
    >
      <!---->
      
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <!---->
        <!---->
        
        <div
          class="v-row"
        >
          <div
            class="v-col"
          >
            <div
              class="v-avatar v-theme--light v-avatar--density-default v-avatar--variant-flat ma-2 pa-1"
              style="background-color: #333; color: #fff; caret-color: #fff; width: 48px; height: 48px;"
            >
              <div
                aria-label=""
                class="v-responsive v-img v-img--booting"
              >
                <div
                  class="v-responsive__sizer"
                />
                
                
                <transition-stub
                  appear="true"
                  css="true"
                  name="fade-transition"
                  persisted="false"
                >
                  <img
                    alt=""
                    class="v-img__img v-img__img--cover"
                    src="/src/assets/it4c-logo2-clean-bg_alpha-128x128.png"
                    style="display: none;"
                  />
                </transition-stub>
                <transition-stub
                  appear="false"
                  css="true"
                  name="fade-transition"
                  persisted="false"
                >
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                
                
                <!---->
              </div>
              
              <!---->
              <span
                class="v-avatar__underlay"
              />
              
            </div>
          </div>
          <div
            class="v-col d-flex align-center justify-center grow"
          >
            <a
              class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-flat"
              href="/"
            >
              
              <span
                class="v-btn__overlay"
              />
              <span
                class="v-btn__underlay"
              />
              
              <!---->
              <span
                class="v-btn__content"
                data-no-activator=""
              >
                
                
                
                $t('menu.home')
                
                
                
              </span>
              <!---->
              <!---->
            </a>
            <a
              class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-flat"
              href="/app"
            >
              
              <span
                class="v-btn__overlay"
              />
              <span
                class="v-btn__underlay"
              />
              
              <!---->
              <span
                class="v-btn__content"
                data-no-activator=""
              >
                
                
                
                $t('menu.app')
                
                
                
              </span>
              <!---->
              <!---->
            </a>
            <a
              class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-flat"
              href="/about"
            >
              
              <span
                class="v-btn__overlay"
              />
              <span
                class="v-btn__underlay"
              />
              
              <!---->
              <span
                class="v-btn__content"
                data-no-activator=""
              >
                
                
                
                $t('menu.about')
                
                
                
              </span>
              <!---->
              <!---->
            </a>
          </div>
          <div
            class="v-col"
          >
            <div
              class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-switch d-flex justify-end mr-5"
            >
              <!---->
              <div
                class="v-input__control"
              >
                
                <div
                  class="v-selection-control v-selection-control--dirty v-selection-control--density-default"
                >
                  <div
                    class="v-selection-control__wrapper text-success"
                  >
                    
                    <div
                      class="v-switch__track bg-success"
                    >
                      <!---->
                      <!---->
                    </div>
                    
                    <div
                      class="v-selection-control__input"
                    >
                      
                      
                      <input
                        aria-describedby="switch-4-messages"
                        aria-disabled="false"
                        aria-label="$t('language.german')"
                        id="switch-4"
                        type="checkbox"
                        value="true"
                      />
                      <div
                        class="v-switch__thumb bg-success"
                      >
                        <transition-stub
                          appear="false"
                          css="true"
                          name="scale-transition"
                          persisted="false"
                        >
                          <!---->
                        </transition-stub>
                      </div>
                      
                      
                    </div>
                  </div>
                  <label
                    class="v-label v-label--clickable"
                    for="switch-4"
                  >
                    <!---->
                    
                    $t('language.german')
                    
                  </label>
                </div>
                
              </div>
              <!---->
              <div
                class="v-input__details"
              >
                <transition-group-stub
                  appear="false"
                  aria-live="polite"
                  class="v-messages"
                  css="true"
                  id="switch-4-messages"
                  name="slide-y-transition"
                  persisted="false"
                  role="alert"
                  tag="div"
                >
                  <!---->
                </transition-group-stub>
                <!---->
              </div>
            </div>
          </div>
        </div>
        
        <!---->
      </div>
      
      
      <transition-stub
        appear="false"
        css="true"
        name="expand-transition"
        persisted="false"
      >
        <!---->
      </transition-stub>
      
    </header>
    
  </div>
</div>
`;
